sycl/source/function_pointer.cpp

```diff
@@ -16,13 +16,19 @@ namespace detail {
 ext::oneapi::device_func_ptr_holder_t
 getDeviceFunctionPointerImpl(device &D, program &P, const char *FuncName) {
   ext::oneapi::device_func_ptr_holder_t FPtr = 0;
-  // FIXME: return value must be checked here, but since we cannot yet check
-  // if corresponding extension is supported, let's silently ignore it here.
+
   const detail::plugin &Plugin = detail::getSyclObjImpl(P)->getPlugin();
-  Plugin.call<cl::sycl::detail::PiApiKind::piextGetDeviceFunctionPointer>(
+  pi_result Result = Plugin.call_nocheck<
+      cl::sycl::detail::PiApiKind::piextGetDeviceFunctionPointer>(
       detail::pi::cast<pi_device>(detail::getSyclObjImpl(D)->getHandleRef()),
       detail::pi::cast<pi_program>(detail::getSyclObjImpl(P)->getHandleRef()),
       FuncName, &FPtr);
+  // If extension is not supported fallback method is used which returns only
+  // if the function exists or not. So, the address is not valid therfore return
+  // no address
+  if (Result == PI_FALLBACK_SUCCESS || Result == PI_FALLBACK_FAILURE)
+    FPtr = 0;
```

Comment:
It's not clear to me why it is OK to return "0" address from this function? Aren't its users expecting valid address? Instead  there should probably be added a new function that returns boolean saying if functions was found or not.

Reply:
Yes, sorry this was an oversight. I will change the code so it throws an with a message indicating that this failed, and if the function existed or not.